### PR TITLE
Fix bug #1435 (data loss when dumping payloads to JSON) v2

### PR DESCRIPTION
**Pull request by gozzy**
Slightly fixed #1423.

This pull request tries to fix bug #1435. Data should not be lost any more.

### CODE REVIEW

File: src/output-json-alert.c [hunk 1]
```diff
@@ -93,12 +93,13 @@ typedef struct JsonAlertLogThread_ {
...
  * so we can report them in JSON output. */
-static int AlertJsonPrintStreamSegmentCallback(const Packet *p, void *data, uint8_t *buf, uint32_t buflen)
+static int AlertJsonDumpStreamSegmentCallback(const Packet *p, void *data, uint8_t *buf, uint32_t buflen)
 {
     MemBuffer *payload = (MemBuffer *)data;
 
-    PrintStringsToBuffer(payload->buffer, &payload->offset, payload->size,
-                         buf, buflen);
+    // We can't user Print...ToBuffer functions as we need real raw data.
+    memcpy(payload->buffer + payload->offset, buf, buflen);
```
Comment (inliniac) [inline src/output-json-alert.c]: It looks like the PrintStringsToBuffer function that is replaced here does check the size of payload->buffer (quite indirectly in the PrintBufferData macro). Is it safe to unconditionally (and without boundary checking) copy data to payload->buffer here?

Comment (gozzy) [inline src/output-json-alert.c]: I think it'll be better to use payload->size instead of buflen. Payload is 4k, but who knows if it's enough.

Comment (gozzy) [inline src/output-json-alert.c]: Or we should truncate buflen if it's greater than payload->size

Comment (inliniac) [inline src/output-json-alert.c]: I think I would prefer a 'safe' wrapper for memcmy in the 'Buffer API'.

Comment (gozzy) [inline src/output-json-alert.c]: You mean MemBufferWriteRaw?

**Comment by gozzy** [comment]
Replaced with #1464

**Pull request closed without merging**